src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.storage.e2e.spec.ts

```diff
@@ -0,0 +1,206 @@
+
+      const collection = await assertEventStoreSetUpCollection(
+        client.db(),
+        customCollectionName,
+      );
+      const stream = await collection.findOne();
+      assertIsNotNull(stream);
+    } finally {
+      await client.close();
+    }
+  });
+
+  void it('sets up database and collection with default SINGLE_COLLECTION', async () => {
+    const client = new MongoClient(mongodb.getConnectionString(), {
+      directConnection: true,
+    });
+
+    try {
+      const eventStore = getMongoDBEventStore({
+        storage: {
+          type: 'SINGLE_COLLECTION',
+        },
+        client,
+      });
+
+      await assertCanAppend(eventStore);
+
+      const collection = await assertEventStoreSetUpCollection(
+        client.db(),
+        DefaultMongoDBEventStoreCollectionName,
+      );
+      const stream = await collection.findOne();
+      assertIsNotNull(stream);
+    } finally {
+      await client.close();
+    }
+  });
+
+  void it('sets up database and collection with CUSTOM colleciton resolution', async () => {
+    const customCollectionSuffix = uuid();
+    const databaseName = uuid();
+
+    const client = new MongoClient(mongodb.getConnectionString(), {
+      directConnection: true,
+    });
+
+    try {
+      const eventStore = getMongoDBEventStore({
+        storage: {
+          type: 'CUSTOM',
+          collectionFor: (
+            streamType: string,
+          ): MongoDBEventStoreCollectionResolution => ({
+            collectionName: `${streamType}:${customCollectionSuffix}`,
+            databaseName,
+          }),
+        },
+        client,
+      });
+
+      await assertCanAppend(eventStore);
+
+      const collection = await assertEventStoreSetUpCollection(
+        client.db(databaseName),
+        `${streamType}:${customCollectionSuffix}`,
+      );
+      const stream = await collection.findOne();
+      assertIsNotNull(stream);
+    } finally {
+      await client.close();
+    }
+  });
+});
+
+const assertCanAppend = async (eventStore: MongoDBEventStore) => {
+  const productItem: PricedProductItem = {
+    productId: '123',
+    quantity: 10,
+    price: 3,
+  };
+  const shoppingCartId = uuid();
+  const streamName = toStreamName(streamType, shoppingCartId);
+
+  const result = await eventStore.appendToStream<ShoppingCartEvent>(
+    streamName,
+    [{ type: 'ProductItemAdded', data: { productItem } }],
+    { expectedStreamVersion: STREAM_DOES_NOT_EXIST },
+  );
+
+  assertNotEqual(
+    result.nextExpectedStreamVersion,
+    MongoDBEventStoreDefaultStreamVersion,
+  );
+};
+
+const assertEventStoreSetUpCollection = async (
+  db: Db,
+  collectionName: string,
+): Promise<Collection<EventStream<ProductItemAdded | DiscountApplied>>> => {
+  const existingCollections = await db.collections();
+  const collection = existingCollections.find(
+    (c) => c.collectionName,
+    collectionName,
+  );
+
+  assertOk(collection);
+
+  const indexes = await collection.indexes();
```

Comment:
**COMMENT:** I'm testing indexes in e2e, as I want to ensure they're setup. Testing that in the unit tests would require too much mambo jumbo around mocking, which I don't like.